### PR TITLE
Refactoring Code

### PR DESCRIPTION
**Pull request by Ashnaali3255**
### In this project we did the following:

- made sure YAGNI, KISS, and DRY rules and Ruby style guide are respected.
- followed SOLID principles.
- pay special attention to the "Single-responsibility Principle".
- made sure that the role of main.rb is only to start our code and ask the user for the initial choice of options (or to quit).
- extracted some pieces of logic into separate class(es).

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -26,10 +26,6 @@ def correct_name
     @name
   end
 
-  def add_rental(date, book)
-    Rental.new(date, book, self)
-  end
-
```
Comment (mckent05) [inline person.rb]: - Please add back this method you implemented from the previous lessons 🙏

File: book.rb [hunk 2]
```diff
@@ -6,8 +6,4 @@ def initialize(title, author)
     @author = author
     @rentals = []
   end
-
-  def add_rental(date, person)
-    Rental.new(date, self, person)
-  end
```
Comment (mckent05) [inline book.rb]: - Please add back this method you implemented from the previous lessons 🙏

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @Ashnaali3255, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job refactoring your app 👏 


## Required Changes ♻️
- You have done great so far, but you have not properly refactored your app. Remember, the goal of the SOLID principle is to make your codebase more maintainable and prepare it for future growth. According to the Single responsibility rule, please refactor your `app.rb` file. I suggest you extract some pieces of code into different files or classes. E.g. You can have one file that is responsible for creating a Person either Student or Teacher. There could be another file for creating a book etc. Please fix this 👍 👍 
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sumairq — APPROVED** [review]
### Hi Team , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**